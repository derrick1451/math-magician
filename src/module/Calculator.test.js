/** @format */

import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import PropTypes from 'prop-types';
import Calculate from './Calculator';

const Button = ({ onClick, children }) => (
  <button type="button" onClick={onClick}>{children}</button>
);

describe('Calculator page component <Calculator />', () => {
  test('Snapshot of the home page', () => {
    const component = renderer.create(<Calculate />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('calls onClick prop when clicked', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>AC</Button>);
    fireEvent.click(screen.getByText(/AC/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};
