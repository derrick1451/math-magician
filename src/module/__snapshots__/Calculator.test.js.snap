// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page component <Calculator /> Snapshot of the home page 1`] = `
<div
  className="container"
>
  <h2>
    Let’s do some math
  </h2>
  <div
    className="wrapper"
  >
    <div
      className="screen"
    >
      0
    </div>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operator color-change"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operator color-change"
      onClick={[Function]}
      type="button"
    >
      *
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operator color-change"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operator color-change"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="number zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operator color-change"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
